### PR TITLE
Reworking the handling of overlaps

### PR DESCRIPTION
**Pull request by sloriot**
There were a couple of cases that were not working correctly due to overlaps.

### CODE REVIEW

File: Sweep_line_2/include/CGAL/Sweep_line_2/Sweep_line_event.h [hunk 1]
```diff
@@ -175,6 +204,10 @@ class Sweep_line_event {
   void push_back_curve_to_left(Subcurve* curve)
   { m_leftCurves.push_back(curve); }
 
+  /*! Add a subcurve to the container of right curves (without checks). */
+  void push_back_curve_to_right(Subcurve* curve)
+  { m_rightCurves.push_back(curve); }
+
```
Comment (sloriot) [inline Sweep_line_2/include/CGAL/Sweep_line_2/Sweep_line_event.h]: Check if it is used

File: Sweep_line_2/include/CGAL/Sweep_line_2/Sweep_line_subcurve.h [hunk 2]
```diff
@@ -111,6 +112,11 @@ class Sweep_line_subcurve {
   /*! Set the last intersecing curve so far. */
   void set_last_curve(const X_monotone_curve_2& cv) { m_lastCurve = cv; }
 
+  /*! Check if the given event is the matches the left-end event. */
+  template <typename SweepEvent>
+  bool is_start_point(const SweepEvent* event) const
+  { return (m_left_event == (Event*)event); }
+
```
Comment (sloriot) [inline Sweep_line_2/include/CGAL/Sweep_line_2/Sweep_line_subcurve.h]: Check it is used

**Comment by lrineau** [comment]
What is the status of this PR? Should it be tested tonight?

**Comment by sloriot** [comment]
It is under review by @efifogel. No need to test it for now, previous test results were fine.

**Comment by lrineau** [comment]
Any update since https://github.com/CGAL/cgal/pull/2342#issuecomment-326974852?

**Comment by lrineau** [comment]
Sorry for the close/reopen. Wrong button on the Github web UI.

**Comment by sloriot** [comment]
On my TODO list but busy right now.

**Comment by sloriot** [comment]
The branch should be ready now!

**Pull request merged**